src/components/features/ViewportListener/ViewportListener.js

```diff
@@ -0,0 +1,46 @@
+import { useEffect, useLayoutEffect, useCallback } from 'react';
+import PropTypes from 'prop-types';
+import { setViewport } from '../../../redux/viewportRedux';
+import { connect } from 'react-redux';
+
+const ViewportListener = ({ setViewport }) => {
+  const breakpoints = {
+    tablet: 992,
+    mobile: 768,
+  };
+
+  const resizeHandler = useCallback(() => {
+    if (window.innerWidth < breakpoints.mobile) {
+      return 'mobile';
+    } else if (
+      window.innerWidth >= breakpoints.mobile &&
+      window.innerWidth < breakpoints.tablet
+    ) {
+      return 'tablet';
+    } else {
+      return 'desktop';
+    }
+  }, [breakpoints.mobile, breakpoints.tablet]);
+
+  useEffect(() => {
+    setViewport(resizeHandler());
+  }, [resizeHandler, setViewport]);
+
+  useLayoutEffect(() => {
+    window && window.addEventListener('resize', () => setViewport(resizeHandler()));
+    return () =>
+      window.removeEventListener('resize', () => setViewport(resizeHandler()));
```

Comment:
Aby działała metoda `removeEventListener` nie możesz korzystać z anonimowych funkcji, musisz stworzyć funkcję np `handleResize()` i ją wykorzystać przy rejestracji obsługi zdarzenia jak przy usuwaniu obsługi zdarzenia

```
 window.addEventListener('resize', this.handleResize);
 window.removeEventListener('resize', this.handleResize);
```

Reply:
Czy w przypadku komponentu funkcyjnego kontekst this jest potrzebny?

Reply:
Nie, wrzuciłem przykładowo